org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/ProjectUtils.java

```diff
@@ -581,4 +565,21 @@ public static int getMaxProjectProblemSeverity() {
 		return maxSeverity;
 	}
 
+	private static boolean isBuildFileMarker(IMarker marker, IProject project) {
```

Comment:
Is there any case that the marker is attached to instances of `IProject`?

Reply:
We just need the type of IProject here. e.g., `ProjectUtils.isMavenProject(project)`.

Reply:
Ok `maxSeverity = Math.max(maxSeverity, project.findMaxProblemSeverity(null, true, IResource.DEPTH_ZERO));` is used for project level markers, then it makes sense.